cli/azd/pkg/apphost/generate.go

```diff
@@ -262,6 +270,10 @@ func newInfraGenerator() *infraGenerator {
 			DaprComponents:                  make(map[string]genDaprComponent),
 			CosmosDbAccounts:                make(map[string]genCosmosAccount),
 			SqlServers:                      make(map[string]genSqlServer),
+			InputParameters:                 make(map[string]genInputParameter),
+			BicepModules:                    make(map[string]genBicepModules),
+			OutputParameters:                make(map[string]genOutputParameter),
```

Comment:
I'm not sure if it would be cleaner to have a single `OutputParameters` map and have `genOutputParameter` have a member that says if it is a secret or not.  I feel like there might be value in having all the outputs in a single map but I could be wrong.  It does feel like the "secretness" is better tied to the output itself instead of being a property inferred based on what map we place it in.

Reply:
My preference is always using just one struct and a field for `isSecret`.
The only reason I added to maps here was to follow the same pattern we decided before for handling the `env` map, where a env key-values can be either secret or non-secret.

I think we can re-evaluate the approach and move from `map[string]string` to represent the environment to `map[string]genEnvVar`.

Since the pattern we select impacts how we iterate and write the values on the go-templates, I would like to keep it the same for all cases.